proto/aserto/directory/model/v3/model.proto

```diff
@@ -7,43 +7,50 @@ option go_package = "github.com/aserto-dev/go-directory/aserto/directory/model/v
 import "google/api/field_behavior.proto";
 import "google/protobuf/empty.proto";
 
-import "aserto/directory/common/v3/common.proto";
-
 service Model {
-    rpc GetManifest(GetManifestRequest) returns (stream GetManifestResponse) {};
-    rpc SetManifest(stream SetManifestRequest) returns (SetManifestResponse) {};
+    rpc GetModel(GetModelRequest) returns (stream GetModelResponse) {};
+    rpc SetModel(stream SetModelRequest) returns (SetModelResponse) {};
 }
 
-message GetManifestRequest {
+message Metadata {
     // manifest name (unique, lc-string)
-    string name                                                 = 1     [(google.api.field_behavior) = REQUIRED];
+    string  name                      = 1;
     // manifest version (semver notation)
-    string version                                              = 2     [(google.api.field_behavior) = REQUIRED];
+    string  version                   = 2;
 }
 
-message GetManifestResponse {
-    // manifest instance
-    aserto.directory.common.v3.Manifest result                  = 1;
+enum OnDelete {
+    ON_DELETE_UNKNOWN                  = 0x0;  // default, will be treated as fail
+    ON_DELETE_FAIL                     = 0x1;  // fail if a type requires to be deleted
+    ON_DELETE_DELETE                   = 0x2;  // delete if a types is required based on a new model
+    ON_DELETE_NUKE                     = 0x3;  // delete all types before creating the new ones
 }
 
-message SetManifestRequest {
-    // manifest instance
-    aserto.directory.common.v3.Manifest manifest                = 1     [(google.api.field_behavior) = REQUIRED];
+message MetadataReq {
+    Metadata metadata                 = 1       [(google.api.field_behavior) = REQUIRED];
+    // operation code indicating how to handle types deletion
+    OnDelete op_code                  = 2       [(google.api.field_behavior) = REQUIRED];
 }
 
-message SetManifestResponse {
-    // manifest instance
-    aserto.directory.common.v3.Manifest result                  = 1;
+message GetModelRequest {
+    google.protobuf.Empty result      = 1;    // empty result
 }
 
-message DeleteManifestRequest {
```

Comment:
Not having a separate delete method breaks with the existing pattern the rest of the directory APIs use of Get/Set/Delete